### PR TITLE
Tilt metrics fix + docs update

### PR DESCRIPTION
**Pull request by hrak**
*Issue #, if available:*

*Description of changes:*

This PR fixes an issue of CAPC failing to start with the following error due to double injection of prometheus metrics settings (this is [taken care of by tilt-prepare now](https://github.com/kubernetes-sigs/cluster-api/blob/38a968ff38ac1d9075dd8832e5ecb820ddb3c45f/hack/tools/tilt-prepare/main.go#L752))

`Build Failed: Deployment.apps "capc-controller-manager" is invalid: spec.template.spec.containers[0].ports[2].name: Duplicate value: "metrics"`

It also updates the tilt docs to remove old info regarding credentials, and adds an example on how to enable a CAPI feature gate.

*Testing performed:*

make tilt-up successfully starts CAPC.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by hrak** [comment]
> Setting up environment failed

Can someone please check this bot?

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -28,7 +28,6 @@ patchesStrategicMerge:
 - manager_image_patch_edited.yaml
 - manager_webhook_patch.yaml
 - webhookcainjection_patch.yaml
-- manager_prometheus_metrics_patch.yaml
```
Comment (vishesh92) [inline config/default/kustomization.yaml:31]: This won't lead to prometheus metrics not getting exposed during deployment of CAPC (without tilt). right?

Comment (hrak) [reply config/default/kustomization.yaml:31]: Yes it does. I've done a little round across other providers ([CAPI](https://github.com/kubernetes-sigs/cluster-api/blob/main/config/default/kustomization.yaml), [CABPK](https://github.com/kubernetes-sigs/cluster-api/blob/main/bootstrap/kubeadm/config/default/kustomization.yaml) [CAPO](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/config/default/kustomization.yaml) [CAPD](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/infrastructure/docker/config/default/kustomization.yaml) and none of them do this in the released manifest. [CAPA](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/config/default/kustomization.yaml) took a different approach and [creates a little helper service ](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/config/default/metrics_service.yaml).

**Comment by hrak** [comment]
Just pushed a small additional change that allows Tilt and e2e tests to replace the metrics listen address (and removes a hack that made this work in e2e before)

related: https://github.com/kubernetes-sigs/cluster-api/commit/7768a6db2ee986da2458e9c73d4c3c988095a4a7#diff-df1234d6ebfa1846a2bace68630ce653b3402bd4388aa4b0b8fb724df34b4af7

File: docs/book/src/development/tilt.md [hunk 2]
```diff
@@ -35,13 +35,23 @@ Next, create a `tilt-settings.json` file and place it in your local copy of `clu
...
+**Example `tilt-settings.json` for CAPC clusters with experimental feature gate:**
+
 ```json
 {
     "default_registry": "gcr.io/your-project-name-here",
     "provider_repos": ["../cluster-api-provider-cloudstack"],
     "enable_providers": ["kubeadm-bootstrap", "kubeadm-control-plane", "cloudstack"],
     "kustomize_substitutions": {
-        "CLOUDSTACK_B64ENCODED_CREDENTIALS": "RANDOM_STRING==",
+        "EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION": "true",
```
[the remark this answers is not in the document]
Comment (vishesh92) [reply docs/book/src/development/tilt.md:54]: https://cluster-api.sigs.k8s.io/tasks/experimental-features/ignition#experimental-feature-ignition-bootstrap-config-alpha

**Pull request merged**